common.eai

```diff
@@ -12857,31 +12859,31 @@ function SingleMeleeAttackAM takes boolean needs_exp, boolean has_siege, boolean
 
       // take expansions as needed
       //
-      if needs_exp then
-        if first_expansion_chosen and (current_expansion == null or CheckExpansionTaken(current_expansion)) then
+      if needs_exp and first_expansion_chosen then
+        if current_expansion == null or CheckExpansionTaken(current_expansion) then
           call ChooseExpansion()
-        endif
-        if current_expansion == null then
-            call Trace("Need expansion but none found")
-        else
-          set creep = GetExpFoe(creep)
-          if creep != null then
-            call Trace("Creep the Expansion")
-            set attacking_expansion = true
-            set battle_radius = creep_battle_radius
-            call SetAllianceTargetIfLeadAndChat(creep, C_Expansion)
-            call FormGroupAM(3)
-            call AttackMoveKillCreep(creep)
-            call ClearAllianceTarget(creep)
-            if creep == null or not UnitAlive(creep) then
-              set take_exp = false
-            endif
-            call Sleep(10)
-            set attacking_expansion = false
-            set creep = null
+          if current_expansion == null or CheckExpansionTaken(current_expansion) then
+            call Trace("Need expansion but can not expansion or none found")
             return
```

Comment:
This return statement will break late game play, as if it wants a mine but there aren't any it will no longer attack players or creep.

I'm going to merge anyway as can fix afterwards.